Source/FieldSolver/FiniteDifferenceSolver/MacroscopicEvolveM.cpp

```diff
@@ -138,22 +147,36 @@ void FiniteDifferenceSolver::MacroscopicEvolveM (
                 + dt * Gil_damp * ( M_xface_prev(i, j, k, 0) * (M_xface_prev(i, j, k, 2) * Hx_eff - M_xface_prev(i, j, k, 0) * Hz_eff)
                 - M_xface_prev(i, j, k, 1) * ( M_xface_prev(i, j, k, 1) * Hz_eff - M_xface_prev(i, j, k, 2) * Hy_eff));
 
-
               // temporary normalized magnitude of M_xface field at the fixed point
               // re-investigate the way we do Ms interp, in case we encounter the case where Ms changes across two adjacent cells that you are doing interp
-              amrex::Real mag_normalized = std::sqrt( std::pow(M_xface(i, j, k, 0),2.0) + std::pow(M_xface(i, j, k, 1),2.0) +
+              amrex::Real M_magnitude_normalized = std::sqrt( std::pow(M_xface(i, j, k, 0),2.0) + std::pow(M_xface(i, j, k, 1),2.0) +
                       std::pow(M_xface(i, j, k, 2),2.0) ) / MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr);
 
-              // check the normalized error
-              if ( amrex::Math::abs(1._rt-mag_normalized) > mag_normalized_error ){
-                  printf("i = %d, j=%d, k=%d\n", i, j, k);
-                  printf("mag_normalized = %f, mag_normalized_error=%f\n", mag_normalized, mag_normalized_error);
-                  amrex::Abort("Exceed the normalized error of the M_xface field");
+              if (M_normalization > 0){
+                  // check the normalized error
+                  if (amrex::Math::abs(1._rt - M_magnitude_normalized) > mag_normalized_error){
+                      printf("i = %d, j=%d, k=%d\n", i, j, k);
+                      printf("M_magnitude_normalized = %f, mag_normalized_error=%f\n", M_magnitude_normalized, mag_normalized_error);
+                      amrex::Abort("Exceed the normalized error of the M_xface field");
+                  }
+                  // normalize the M_xface field
+                  M_xface(i, j, k, 0) /= M_magnitude_normalized;
+                  M_xface(i, j, k, 1) /= M_magnitude_normalized;
+                  M_xface(i, j, k, 2) /= M_magnitude_normalized;
+              }
+              else if (M_normalization == 0){
+                  // check if M magnitude is below Ms for unsaturated material
+                  if (M_magnitude_normalized > 1._rt){
+                      printf("i = %d, j=%d, k=%d\n", i, j, k);
+                      printf("M_magnitude_normalized = %f, Ms = %f \n", M_magnitude_normalized, 
+                          MacroscopicProperties::macro_avg_to_face(i,j,k,amrex::IntVect(1,0,0),mag_Ms_arr));
+                      amrex::Abort("Caution: Unsaturated material has M_xface exceeding the saturation magnetization");
+                  }
+              }
+              else{
+                  printf("M_normalization = %d \n", M_normalization);
```

Comment:
This is certainly not needed since we already checked above.
Hence, my previous comment; maybe we check the validity of M_normalization once at the beginning of the simulation so we don't need these if tests.